transform/mattermost-analytics/models/intermediate/product/licenses/int_licenses_per_source.sql

```diff
@@ -0,0 +1,59 @@
+-- Temporarily materialize during development
+{{
+    config({
+        "materialized": "table",
+        "snowflake_warehouse": "transform_l"
+    })
+}}
+with stripe_licenses as (
+    select
+        s.license_id,
+        -- Convert to epoch in order to remove timezone
+        to_timestamp_ntz(date_part(epoch_second, license_end_at)) as expire_at
+    from
+        {{ ref('stg_stripe__subscriptions')}} s
+        left join {{ ref('stg_stripe__subscription_items')}} si on s.subscription_id = si.subscription_id
+        left join {{ ref('stg_stripe__products')}} p on si.product_id = p.product_id
+    where
+        -- Onprem subscription/subscription items
+        p.name not ilike '%cloud%'
+        -- Skip support subscription items and focus on main plan
+        and p.name <> 'Premier Support'
+        -- Skip incomplete subscriptions
+        and s.status <> 'incomplete_expired'
+        -- Data before this date might not be in-line with specification
+        and s.created_at > '2021-04-01'
+        -- Ignore invalid records
+        and s.license_id is not null
```

Comment:
maybe also filter out the ones that have month/day duration <= 0

Reply:
If the total number of days is less than a month, `duration_months` is going to be 0. There's also a `duration_days` to allow for different level of granularity.

After checking the data I noticed `duration_months = 0` only applies to three licenses. Two of them have duration of 1 week (7 days), and the other one is 29 but it's a trial.

Reply:
The same reason licenses with excessive duration (there are approx 90 with duration > 36 months are suspect, the ones with small duration are suspect as well. They are recently created ones, so it is a sign of things to come? I'll ping Hanna.